protected/humhub/modules/user/widgets/PeopleFilterPicker.php

```diff
@@ -132,18 +158,38 @@ protected function getItemImage($item)
     public function getSuggestions($keyword = '')
     {
         if (empty($this->defaultResults)) {
-            return User::find()
+            $query = User::find()
                 ->select(['id' => $this->itemKey, 'text' => $this->itemKey])
                 ->visible()
                 ->joinWith('profile')
-                ->andWhere(['LIKE', $this->itemKey, $keyword])
-                ->groupBy($this->itemKey)
+                ->andWhere(['LIKE', $this->itemKey, $keyword]);
+
+            if ($this->query instanceof PeopleQuery && $this->query->isFiltered()) {
+                $query->andWhere(['IN', 'user.id', $this->query->getFilteredUsersSubQuery()]);
```

Comment:
@yurabakhtin I think this approach is too unperformant for larger user bases. 

Can't we somehow clone and extend the existing query instead of using a SubQuery? 

Do you see a possibility to do something like this here?
```sql
SELECT DISTINCT city FROM profile WHERE country = 'DE' /* ... */;
```

Reply:
@luke- Ok, I have done this in commit https://github.com/humhub/humhub/pull/6812/commits/e13e937b83139db202294a2b463279953492bd81.

Reply:
@yurabakhtin Thanks looks better in terms of performance. I still need some time for review. Can you please check the failing tests and also add tests in the meantime.

Reply:
@luke- The tests has been fixed by re-run.
I have started new tests for people filters in the commit https://github.com/humhub/humhub/pull/6812/commits/fd6616b1e9668f760d167133584c6da7b0a4058f, but they are not completed yet, will do later.

Reply:
@luke- I have done tests for reduced filters in commit https://github.com/humhub/humhub/pull/6812/commits/6609e0d9f6b39ae541d65448bd0f252d817629ad.